doc/lua_api.md

```diff
@@ -7772,6 +7772,8 @@ of the `${k}` syntax in formspecs is not deprecated.
     * The range for the value is system-dependent (usually 32 bits).
       The value will be converted into a string when stored.
 * `get_float(key)`: Returns `0` if key not present.
+* `set_bool(key, value)` Can be set to either `true` or `false`. Any other value crashes it.
+* `get_bool(key)`: Gives an error if key not present.
```

Comment:
Actually. It does give an error.

Reply:
Wouldn't it make more sense if:

- `get_bool()` returns `true`, `false`, or `nil` if the key is not present, and
- set_bool() accepts `true` or `false`, and removes the key when given `nil`

Having to use `set_string(key, nil)` as the only way to remove a key is a bit awful...

Reply:
I don't see how making the key nil, is useful. Please explain your idea more.

Reply:
I think they're saying `set_bool(key, nil)` should remove the key, and `get_bool(key)` should return `nil` for an absent key. This would be consistent with how Lua tables work.

Reply:
Sorry, my poor English. What @appgurueu said is **exactly** what I meant. Since Lua can represent the absence of a value with `nil` and we do need to represent a condition in which there is no such a key in the metadata store (or to set this condition), why not to use it?

As a side node and probably off-topic here, this same "convention" should also be implemented in all the others get/set pairs (specifically, `_float` and `_int`). This would be more consistent (and avoid situations like I when I need to remove an int and had to set a `nil` string with the same key). Although that would probably break some existing mods which may just not be checking for `nil` in values returned by `get_` functions.

However, the present case of boolean get/set functions is a fresh implementation, so it's worth to implement it the right way! :-)